testutil/testdevice_linux.go

```diff
@@ -0,0 +1,97 @@
+// +build linux
+
+package testutil
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+type testDevice struct {
+	opts       TestDeviceOpts
+	fileName   string
+	deviceName string
+}
+
+func NewTestDevice(t *testing.T, opts TestDeviceOpts) TestDevice {
+	if opts.Mountpoint == "" || len(opts.MKFS) == 0 {
+		t.Fatalf("invalid opts: %+v", opts)
+	}
+	// create temporary file for the disk image
+	file, err := ioutil.TempFile("", "containerd-testdevice")
+	if err != nil {
+		t.Fatalf("could not create temporary file for test device: %v", err)
+	}
+
+	if err := file.Truncate(opts.Size); err != nil {
+		t.Fatal(err)
+	}
+	file.Close()
+
+	// create device
+	losetup := exec.Command("losetup", "--find", "--show", file.Name())
+	p, err := losetup.Output()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	deviceName := strings.TrimSpace(string(p))
+	t.Logf("Created loop device %s", deviceName)
+
+	// format
+	t.Logf("Creating filesystem with %v", opts.MKFS)
+	mkfs := exec.Command(opts.MKFS[0], append([]string{deviceName}, opts.MKFS[1:]...)...)
```

Comment:
This ordering does not look correct. I would expect device to come after options.

```
SYNOPSIS
       mkfs [options] [-t type] [fs-options] device [size]
```